src/LearnositySdk/Request/Init.php

```diff
@@ -113,6 +113,15 @@ class Init
      */
     public function __construct($service, $securityPacket, $secret, $requestPacket = null, $action = null)
     {
+        $this->requestPassedAsString = false;
+        if (is_string($requestPacket)) {
+            $requestString = $requestPacket;
+            $requestPacket = json_decode($this->requestString);
```

Comment:
Perhaps we should check whether `is_null($requestPacket)`, meaning that the `$requestString` was not valid JSON, and throw an ValidationException.

The call to `validate` just below will catch the problem and stop, but the error message will be confusing https://github.com/Learnosity/learnosity-sdk-php/blob/51b5730a807f34dcee4f9fbaea84416afeb4f01d/src/LearnositySdk/Request/Init.php#L411-L413

Alternatively, maybe we should just change that error to say “... an array or a valid JSON string”.

All things considered, I think the latter is a better idea.

Reply:
Done - the latter it is, cheers